tla/consensus/ccfraft.tla

```diff
@@ -496,7 +496,7 @@ InitMessagesVars ==
     /\ commitsNotified = [i \in Servers |-> <<0,0>>] \* i.e., <<index, times of notification>>
 
 InitServerVars ==
-    /\ currentTerm = [i \in Servers |-> 0]
+    /\ currentTerm = [i \in Servers |-> 2]
     /\ state       = [i \in Servers |-> IF i \in configurations[i][0] THEN Follower ELSE Pending]
```

Comment:
```suggestion
    /\ state       = [i \in Servers |-> IF i \in configurations[i][0] THEN Leader ELSE Pending]
```

Reply:
Are you also going to change `InitReconfigurationVars` to something like the following to guarantee a single leader at startup?

```tla
InitReconfigurationVars ==
    /\ ...
    /\ \E s \in Servers:
        configurations = [i \in Servers |-> 0 :> {s} ]
     \* There is a single leader in the initial configuration, and a subset of the nodes may know about this config.
    /\ Cardinality(Range(configurations)) = 1
```

Also, remember that `MCccfraft` redefines `InitReconfigurationVars`.

Reply:
I've changed this some more, with guidance from @heidihoward, but yes I will make the change everywhere. Longer term I think we will split the membership/status in different variables also.

Reply:
@lemmy we can't just initialise the configurations, that makes them inconsistent with the log. We also cannot initialise the configurations for other nodes that have not yet joined when they have no log. All configurations are discovered through the log.